### ISSUE TITLE
Parameter bounded by PREVIOUS max

### ISSUE THREAD

**Issue by ggggggggg** [lmfit/lmfit-py#651]
### Description
Setting a parameter value, max and min can lead to the parameter being bounded by the PREVIOUS value of max.

###### A Minimal, Complete, and Verifiable example
```
import lmfit
params = lmfit.Parameters()
params.add(name="a",value=0,max=1,min=0)
params["a"].set(value=100,max=1000,min=0)
params.pretty_print()
```
prints
```
Name     Value      Min      Max   Stderr     Vary     Expr Brute_Step
a         1        0     1000     None     True     None     None
```

I expect `a` to have value `100`.

###### Error message:
None

###### Version information
Python: 3.8.2 (default, Feb 27 2020, 19:58:50) 
[Clang 11.0.0 (clang-1100.0.33.17)]

lmfit: 1.0.0, scipy: 1.4.1, numpy: 1.18.0, asteval: 0.9.18, uncertainties: 3.1.2

###### Link(s)
None

**Comment by reneeotten** [comment]
@ggggggggg this was fixed in commit 2b0f7b56d81da80a1b872517b3ef6e40f540c9d5 and is part of the 1.0.1 release. Please update to the latest version and all should be good!

**Comment by ggggggggg** [comment]
Ok.
